### ISSUE TITLE
Re-adding to group resets shape styles?

### ISSUE THREAD

**Issue by vorg** [runemadsen/rune.js#26]
I'm trying to animate rectangles in an animation and i'm trying to modify z-order and put things on top of each other with `rect.addTo(g) ` but when I do that some shape disappear, some get color and then loose it... Can this be some virtualdom bug?

```
function highlightNext() {
  if (nextRect < rects.length) {
    var rect = rects[nextRect++]
    rect.stroke('#FF0000')
    rect.addTo(g) // ???
  }
}
setInterval(highlightNext, 500)
```

What i'm expecting
![giphy](https://user-images.githubusercontent.com/171001/28138824-057bcf88-674a-11e7-9076-d3f35f93b16f.gif)

What i'm getting
![giphy-2](https://user-images.githubusercontent.com/171001/28138828-07a97c38-674a-11e7-947b-8b51b42f299f.gif)

```javascript
var Rune = require('rune.js')

var r = new Rune({
  container: 'body',
  width: 200,
  height: 200,
  debug: true
})

var bg = r.rect(0, 0, 200, 200).fill('#E7E8E9').stroke(false)

var rects = []
var nextRect = 0
var g = r.group()
rects.push(r.rect(10, 10, 100, 100, g).fill(false).strokeWidth(4))
rects.push(r.rect(30, 30, 100, 100, g).fill(false).strokeWidth(4))
rects.push(r.rect(50, 50, 100, 100, g).fill(false).strokeWidth(4))

function highlightNext() {
  if (nextRect < rects.length) {
    var rect = rects[nextRect++]
    rect.stroke('#FF0000')
    rect.addTo(g) // <----- troublesome line
  }
}

r.play()
setInterval(highlightNext, 500)
```

**Comment by runemadsen** [comment]
Thanks for the report! I'm not sure exactly what's going on, but you shouldn't be adding the rectangles to the group again and again. Here's an example where you animation is working:
https://codepen.io/runemadsen/pen/qjgEdg?editors=0010

I'll take a look at the weird disappearing rectangle.

**Comment by vorg** [comment]
Is there a better way to change z-order then? I want highlighted rects to be on top of the black ones.

**Comment by runemadsen** [comment]
Ah, ok. I think this must be a bug. You should be able to call `remove()` and `add()` on the group to make the shape first in order, but it doesn't seem to be happening. Will look at it tomorrow.

**Comment by runemadsen** [comment]
Okay, I finally had time to jump into this, and I just published version `1.1.0` that fixes this bug. I'll work on adding helper function that can make it easier to change the layering of shapes.
